sdk/test-utils/perfstress/src/testProxyHttpClient.ts

```diff
@@ -114,7 +116,7 @@ export class TestProxyHttpClient {
   async modifyRequest(request: PipelineRequest): Promise<PipelineRequest> {
     if (this._recordingId && (this._mode === "record" || this._mode === "playback")) {
       request = this.redirectRequest(request, this._recordingId);
-      request.allowInsecureConnection = true;
+      request.allowInsecureConnection = this._uri.startsWith("http") || this.insecure;
```

Comment:
```suggestion
      request.allowInsecureConnection = this.insecure;
```

SSL validation should only apply to HTTPS uris, so I think we can just set this based on the "insecure" option.

Reply:
This is from core-rest-pipeline.
Required while using HTTP, 
![image](https://user-images.githubusercontent.com/10452642/135362247-da8e50ad-7241-4910-a113-e32e003d6513.png)

Reply:
Origin https://github.com/Azure/azure-sdk-for-js/pull/14316

Reply:
OK, I think I understand now, but I still think the existing code is incorrect.  I think the `allowInsecureConnection` and `rejectUnauthorized` settings are completely independent.  The former allows core-rest-pipeline to handle `http:` URIs, while the latter allows `https:` URIs with invalid (self-signed) certs.

So I think the correct code is this.  Note the URI needs to start with `http:` (with a trailing `:`), since both `http:` and `https:` start with `http`.

```suggestion
      request.allowInsecureConnection = this._uri.startsWith("http:");
```

Reply:
Or, we could just keep the previous code which always sets `allowInsecureConnect = true`, though it would technically be unnecessary for `https:` URIs.

Reply:
Oops, overlook, makes sense.
Updating.